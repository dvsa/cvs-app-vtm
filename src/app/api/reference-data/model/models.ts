export * from './reference-data.model';
