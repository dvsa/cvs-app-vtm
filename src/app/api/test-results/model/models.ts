export * from './carCancelled';
export * from './carCancelledAdditionalInformation';
export * from './carCancelledAdditionalInformationLocation';
export * from './carCancelledCustomDefects';
export * from './carCancelledDefects';
export * from './carCancelledModType';
export * from './carCancelledTestTypes';
export * from './carCancelledVehicleClass';
export * from './carSubmitted';
export * from './carSubmittedAdditionalInformation';
export * from './carSubmittedDefects';
export * from './carSubmittedTestTypes';
export * from './completeTestResults';
export * from './completeTestResultsVehicleClass';
export * from './customDefect';
export * from './customDefects';
export * from './defect';
export * from './defectAdditionalInformation';
export * from './defectAdditionalInformationLocation';
export * from './defects';
export * from './hgvCancelled';
export * from './hgvSubmitted';
export * from './lgvCancelled';
export * from './lgvSubmitted';
export * from './motorcycleCancelled';
export * from './motorcycleSubmitted';
export * from './motorcycleSubmittedVehicleClass';
export * from './msUserDetails';
export * from './psvCancelled';
export * from './psvCancelledModType';
export * from './psvCancelledTestTypes';
export * from './psvSubmitted';
export * from './psvSubmittedAdditionalInformation';
export * from './psvSubmittedAdditionalInformationLocation';
export * from './psvSubmittedDefects';
export * from './psvSubmittedTestTypes';
export * from './testHistory';
export * from './testResultPutBody';
export * from './testResults';
export * from './testTypeRecords';
export * from './testTypeResults';
export * from './testTypeResultsModType';
export * from './testTypes';
export * from './trlCancelled';
export * from './trlCancelledTestTypes';
export * from './trlSubmitted';
export * from './trlSubmittedTestTypes';
