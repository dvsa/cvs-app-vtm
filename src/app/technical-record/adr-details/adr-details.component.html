<!--  <dl class="technical-record-sub-details govuk-summary-list adr-details">-->
<!--    <div class="govuk-summary-list__row">-->
<!--      <dt class="govuk-summary-list__key">-->
<!--        Able to carry dangerous goods-->
<!--        <span *ngIf="showCheck">-->
<!--          <div class="govuk-radios govuk-radios&#45;&#45;inline" id="radioADR">-->
<!--            <div class="govuk-radios__item">-->
<!--              <input class="govuk-radios__input" id="hasADR" name="hasADR" type="radio" [checked]="!isAdrNull"-->
<!--                     value="true" (change)="switchAdrDisplay($event)">-->
<!--              <label class="govuk-label govuk-radios__label" for="hasADR">-->
<!--                Yes-->
<!--              </label>-->
<!--            </div>-->
<!--            <div class="govuk-radios__item">-->
<!--              <input class="govuk-radios__input" id="hasNotADR" name="hasADR" type="radio"-->
<!--                     [checked]="isAdrNull" value="false" (change)="switchAdrDisplay($event)">-->
<!--              <label class="govuk-label govuk-radios__label" for="hasNotADR">-->
<!--                No-->
<!--              </label>-->
<!--            </div>-->
<!--          </div>-->
<!--        </span>-->
<!--      </dt>-->

<!--      <dd id="test-ableToCarry" class="govuk-summary-list__value">-->
<!--        <div *ngIf="!showCheck">-->
<!--          <span> {{ activeRecord.adrDetails == null ? "No" : "Yes"  }}</span>-->
<!--        </div>-->
<!--      </dd>-->
<!--      <dd class="govuk-summary-list__value"></dd>-->
<!--    </div>-->

<!--    <vtm-adr-details-view>-->
<!--        [activeRecord]="activeRecord"-->
<!--        [techRecordsJson]="techRecordsJson"-->
<!--        [hideForm]="hideForm">-->
<!--    </vtm-adr-details-view>-->

<!--&lt;!&ndash;    <ng-template #adrForm>&ndash;&gt;-->
<!--&lt;!&ndash;        <vtm-adr-details-form *ngIf="!hideForm"&ndash;&gt;-->
<!--&lt;!&ndash;          [vehicletypes$]="vehicletypes$"&ndash;&gt;-->
<!--&lt;!&ndash;          [permittedDangerousGoodsFe$]="permittedDangerousGoodsFe$"&ndash;&gt;-->
<!--&lt;!&ndash;          [guidanceNotesFe$]="guidanceNotesFe$"&ndash;&gt;-->
<!--&lt;!&ndash;          [initialAdrDetails]="activeRecord?.adrDetails">&ndash;&gt;-->
<!--&lt;!&ndash;        </vtm-adr-details-form>&ndash;&gt;-->
<!--&lt;!&ndash;    </ng-template>&ndash;&gt;-->

<!--  </dl>-->
