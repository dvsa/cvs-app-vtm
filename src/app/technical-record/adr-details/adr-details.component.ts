import {Component, OnInit, ViewEncapsulation, ChangeDetectionStrategy, Input} from '@angular/core';
import {Observable} from 'rxjs';
import {select, Store} from '@ngrx/store';
import {IAppState} from '@app/technical-record/adr-details/adr-details-form/store/adrDetailsForm.state';

@Component({
  selector: 'vtm-adr-details',
  templateUrl: './adr-details.component.html',
  styleUrls: ['../../app.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AdrDetailsComponent implements OnInit {

  @Input() activeRecord: any;
  @Input() showCheck: boolean;
  @Input() isAdrNull: boolean;
  @Input() adrData: boolean;
  @Input() hideForm: boolean;
  @Input() techRecordsJson: any;

  vehicletypes$: Observable<string[]>;
  permittedDangerousGoodsFe$: Observable<string[]>;
  guidanceNotesFe$: Observable<string[]>;

  constructor(private _store: Store<IAppState>) { }

  ngOnInit() {
    this.vehicletypes$ = this._store.pipe(select(s => s.vehicleTechRecordModel.vehicleTechRecordModel[0].metadata.adrDetails.vehicleDetails.typeFe));
    this.permittedDangerousGoodsFe$ = this._store.pipe(select(s => s.vehicleTechRecordModel.vehicleTechRecordModel[0].metadata.adrDetails.permittedDangerousGoodsFe));
    this.guidanceNotesFe$ = this._store.pipe(select( s => s.vehicleTechRecordModel.vehicleTechRecordModel[0].metadata.adrDetails.additionalNotes.guidanceNotesFe));
  }

  switchAdrDisplay($event): void {
    this.adrData = !($event.currentTarget.value === 'true');
    this.hideForm = $event.currentTarget.value === 'false';
  }

}
