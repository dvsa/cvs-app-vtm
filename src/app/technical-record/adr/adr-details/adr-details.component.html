<ng-container *ngIf="!edit">
  <h2 class="govuk-heading-m">ADR details</h2>

  <dl class="govuk-summary-list">
    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Vehicle type</dt>
      <dd id="test-ADR-type" class="govuk-summary-list__value">
        {{ adrDetails.vehicleDetails?.type | DefaultNullOrEmpty | titlecase }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Date processed</dt>
      <dd id="test-ADR-approvalDate" class="govuk-summary-list__value">
        {{ adrDetails.vehicleDetails?.approvalDate | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Permitted dangerous goods</dt>
      <dd id="test-ADR-permittedDangerousGoods" class="govuk-summary-list__value">
        <div *ngIf="adrDetails.permittedDangerousGoods; else hyphen">
          <div *ngFor="let perGood of adrDetails.permittedDangerousGoods; let i = index">
            <p>
              <span id="test-ADR-permittedDangerousGoods-{{ i }}">{{ perGood }}</span>
            </p>
          </div>
        </div>
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Compatibility group J</dt>
      <dd id="test-ADR-compatibilityGroupJ" class="govuk-summary-list__value">
        {{ adrDetails?.compatibilityGroupJ | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">Guidance notes</dt>
      <dd id="test-ADR-additionalNotes-number" class="govuk-summary-list__value">
        <div
          *ngIf="
            this.adrDetails.additionalNotes && adrDetails.additionalNotes.number[0];
            else hyphen
          "
        >
          <div *ngFor="let number of adrDetails.additionalNotes.number; let i = index">
            <p>
              <span id="test-ADR-additionalNotes-number-{{ i }}">{{ number }}</span>
            </p>
          </div>
        </div>
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>

    <div class="govuk-summary-list__row">
      <dt class="govuk-summary-list__key">ADR type approval number</dt>
      <dd id="test-ADR-adrTypeApprovalNo" class="govuk-summary-list__value">
        {{ adrDetails.adrTypeApprovalNo | DefaultNullOrEmpty }}
      </dd>
      <dd class="govuk-summary-list__value"></dd>
    </div>
  </dl>
</ng-container>

<ng-template #hyphen><span>-</span></ng-template>

<!-- Edit Applicant Details -->
<ng-container *ngIf="edit">
  <vtm-adr-details-edit [adrDetails]="adrDetails" [metaData]="metaData"></vtm-adr-details-edit>
</ng-container>
