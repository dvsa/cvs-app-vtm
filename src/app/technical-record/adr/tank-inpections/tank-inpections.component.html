<ng-container *ngIf="!edit">
  <div class="govuk-grid-row">
    <div class="govuk-grid-column-full">
      <h2 class="govuk-heading-m">Tank inspections</h2>

      <table class="govuk-table">
        <thead class="govuk-table__head">
          <tr class="govuk-table__row">
            <th scope="col" class="govuk-table__header">Inspection type</th>
            <th scope="col" class="govuk-table__header">Certificate number</th>
            <th scope="col" class="govuk-table__header">Expiry date</th>
          </tr>
        </thead>

        <tbody class="govuk-table__body" *ngIf="showTankHistory; else noTankHistory">
          <tr class="govuk-table__row">
            <td id="test-tc2Type" class="govuk-table__cell">
              {{ tankDetails?.tc2Details?.tc2Type | DefaultNullOrEmpty | titlecase }}
            </td>

            <td id="test-edit-tc2IntermediateApprovalNo" class="govuk-table__cell">
              {{ tankDetails?.tc2Details?.tc2IntermediateApprovalNo | DefaultNullOrEmpty }}
            </td>
            <td id="test-edit-tc2IntermediateExpiryDate" class="govuk-table__cell">
              <span *ngIf="tankDetails?.tc2Details?.tc2IntermediateExpiryDate; else hyphen">
                {{ tankDetails?.tc2Details?.tc2IntermediateExpiryDate | date:'dd/MM/yyyy' }}
              </span>
            </td>
          </tr>

          <tr
            class="govuk-table__row"
            *ngFor="let tc3Details of tankDetails.tc3Details; let i = index"
          >
            <td id="test-tc3Type-{{ i }}" class="govuk-table__cell">
              {{ tc3Details.tc3Type | DefaultNullOrEmpty | titlecase }}
            </td>
            <td id="test-tc3PeriodicNumber-{{ i }}" class="govuk-table__cell">
              {{ tc3Details.tc3PeriodicNumber | DefaultNullOrEmpty }}
            </td>
            <td id="test-tc3PeriodicExpiryDate-{{ i }}" class="govuk-table__cell">
              <span *ngIf="tc3Details.tc3PeriodicExpiryDate; else hyphen">
                {{ tc3Details.tc3PeriodicExpiryDate | date:'dd/MM/yyyy' }}
              </span>
            </td>
          </tr>
        </tbody>

        <ng-template #noTankHistory>
          <tbody class="govuk-table__body">
            <tr class="govuk-table__row">
              <td class="govuk-table__cell">-</td>
              <td class="govuk-table__cell">-</td>
              <td class="govuk-table__cell">-</td>
            </tr>
          </tbody>
        </ng-template>
      </table>
    </div>
  </div>
</ng-container>
<ng-template #hyphen>
  <span>-</span>
</ng-template>

<!-- Edit TANK Inspections -->
<ng-container *ngIf="edit">
  <vtm-tank-inspections-edit
    [tankDetailsData]="tankDetails"
    [vehicleType]="vehicleType"
  ></vtm-tank-inspections-edit>
</ng-container>
