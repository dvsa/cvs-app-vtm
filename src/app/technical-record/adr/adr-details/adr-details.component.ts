import { Component, OnInit, ChangeDetectionStrategy, Input } from '@angular/core';

import { AdrDetails } from '@app/models/adr-details';
import { MetaData } from '@app/models/meta-data';

@Component({
  selector: 'vtm-adr-new-details',
  templateUrl: './adr-details.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class AdrDetailsComponent implements OnInit {
  @Input() edit: boolean;
  @Input() adrDetails: AdrDetails;
  @Input() metaData: MetaData;

  constructor() {}

  ngOnInit() {}
}
