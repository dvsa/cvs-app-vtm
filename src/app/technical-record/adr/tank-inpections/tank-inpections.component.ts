import { Component, ChangeDetectionStrategy, Input, OnChanges } from '@angular/core';

import { AdrDetails } from '@app/models/adr-details';
import { TankDetails } from '@app/models/Tank';

@Component({
  selector: 'vtm-tank-inpections',
  templateUrl: './tank-inpections.component.html',
  changeDetection: ChangeDetectionStrategy.OnPush
})
export class TankInpectionsComponent implements OnChanges {
  showTankHistory: boolean;
  tankDetails: TankDetails;
  vehicleType: string;

  @Input() edit: boolean;
  @Input() adrDetails: AdrDetails;

  constructor() {}

  ngOnChanges(): void {
    const { vehicleDetails } = this.adrDetails;
    this.showTankHistory =
      (vehicleDetails && vehicleDetails.type.includes('tank')) ||
      (vehicleDetails && vehicleDetails.type.includes('battery'));

    this.vehicleType = vehicleDetails && vehicleDetails.type ? vehicleDetails.type : '';

    const { tank } = this.adrDetails;
    this.tankDetails = tank && tank.tankDetails ? tank.tankDetails : undefined;
  }
}
