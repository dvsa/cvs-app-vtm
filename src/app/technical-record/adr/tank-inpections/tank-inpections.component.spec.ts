import { Component, CUSTOM_ELEMENTS_SCHEMA, Input } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import { SharedModule } from '@app/shared';
import { TESTING_UTILS } from '@app/utils/testing.utils';
import { TankDetails, Tank } from '@app/models/Tank';
import { AdrDetails } from '@app/models/adr-details';
import { TankInpectionsComponent } from './tank-inpections.component';

describe('TankInpectionsComponent', () => {
  let component: TestTankInpectionsComponent;
  let fixture: ComponentFixture<TestTankInpectionsComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      imports: [SharedModule],
      declarations: [
        TankInpectionsComponent,
        TestTankInpectionsComponent,
        TestTankInspectionsEditComponent
      ],
      schemas: [CUSTOM_ELEMENTS_SCHEMA]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(TestTankInpectionsComponent);
    component = fixture.componentInstance;
    const tank = {
      ...TESTING_UTILS.mockTank(),
      tankDetails: { tc2Details: TESTING_UTILS.mockTc2Details() } as TankDetails
    } as Tank;

    component.adrDetails = TESTING_UTILS.mockAdrDetails({
      tank: tank
    });
  });

  it('should create view only with populated data', () => {
    component.edit = false;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('should render the editable component if edit is true', () => {
    component.edit = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});

@Component({
  selector: 'test-vtm-tank-inpections',
  template: `
    <vtm-tank-inpections [edit]="edit" [adrDetails]="adrDetails"></vtm-tank-inpections>
  `
})
class TestTankInpectionsComponent {
  edit: boolean;
  adrDetails: AdrDetails;
}

@Component({
  selector: 'vtm-tank-inspections-edit',
  template: `
    <div>{{ tankDetailsData | json }}</div>
  `
})
class TestTankInspectionsEditComponent {
  @Input() tankDetailsData: TankDetails;
}
