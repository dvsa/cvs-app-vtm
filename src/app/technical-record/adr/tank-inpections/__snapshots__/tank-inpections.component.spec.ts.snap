// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TankInpectionsComponent should create view only with populated data 1`] = `
<vtm-tank-inpections
  adrDetails={[Function Object]}
  edit="false"
  showTankHistory={[Function Boolean]}
  tankDetails={[Function Object]}
>
  <div
    class="govuk-grid-row"
  >
    <div
      class="govuk-grid-column-full"
    >
      <h2
        class="govuk-heading-m"
      >
        Tank inspections
      </h2>
      <table
        class="govuk-table table-tank"
      >
        <thead
          class="govuk-table__head"
        >
          <tr
            class="govuk-table__row"
          >
            <th
              class="govuk-table__header"
              scope="col"
            >
              Inspection type
            </th>
            <th
              class="govuk-table__header"
              scope="col"
            >
              Certificate number
            </th>
            <th
              class="govuk-table__header"
              scope="col"
            >
              Expiry date
            </th>
          </tr>
        </thead>
        
        <tbody
          class="govuk-table__body"
        >
          <tr
            class="govuk-table__row"
          >
            <td
              class="govuk-table__cell"
              id="test-tc2Type"
            >
               initial 
            </td>
            <td
              class="govuk-table__cell"
              id="test-tc2IntermediateApprovalNo"
            >
               tank123 
            </td>
            <td
              class="govuk-table__cell"
              id="test-tc2IntermediateExpiryDate"
            >
               18/03/2020 
            </td>
          </tr>
          
        </tbody>
        
      </table>
    </div>
  </div>
</vtm-tank-inpections>
`;

exports[`TankInpectionsComponent should render the editable component if edit is true 1`] = `
<vtm-tank-inpections
  adrDetails={[Function Object]}
  edit={[Function Boolean]}
  showTankHistory={[Function Boolean]}
  tankDetails={[Function Object]}
>
  <vtm-tank-inspections-edit
    ng-reflect-tank-details-data="[object Object]"
  >
    <div>
      {
    "tc2Details": {
      "tc2IntermediateApprovalNo": "tank123",
      "tc2IntermediateExpiryDate": "2020-03-18",
      "tc2Type": "initial"
    }
  }
    </div>
  </vtm-tank-inspections-edit>
</vtm-tank-inpections>
`;
